nova/tests/unit/scheduler/filters/test_external_scheduler_filter.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2024 SAP SE or an SAP affiliate company.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+# Tests for external scheduler filter.
+
+from unittest.mock import MagicMock
+from unittest.mock import patch
+from unittest.mock import sentinel
+
+import nova.conf
+from nova import objects
+from nova.scheduler.filters import external_scheduler_filter
+from nova import test
+from nova.tests.unit.scheduler import fakes
+
+
+CONF = nova.conf.CONF
+
+
+class ExternalSchedulerFilterTestCase(test.NoDBTestCase):
+
+    def setUp(self):
+        super(test.NoDBTestCase, self).setUp()
+        CONF.set_override('external_scheduler_api_url',
+                          'http://127.0.0.1:1234', 'filter_scheduler')
+
+    @patch('requests.post')
+    def test_filter_all(self, mock_post):
+        mock_response = MagicMock()
+        mock_response.status_code = 200
+        mock_response.json.return_value = {
+            'valid_hosts': ['host1', 'host3']
+        }
+        mock_post.return_value = mock_response
+
+        host_attributes = {
+            "status": "up",
+            "vcpus_total": 100,
+            "vcpus_used": 10,
+            "memory_mb": 1000,
+            "memory_mb_used": 10,
+        }
+        all_hosts = [
+            fakes.FakeHostState('host1', 'node1', host_attributes),
+            fakes.FakeHostState('host2', 'node2', host_attributes),
+            fakes.FakeHostState('host3', 'node3', host_attributes)
+        ]
+        instance_request_spec = objects.RequestSpec(
+            context=sentinel.ctx,
+            flavor=objects.Flavor(
+                name="small",
+                vcpus=4,
+                memory_mb=1024
+            ),
+        )
+
+        f = external_scheduler_filter.ExternalSchedulerFilter()
```

Comment:
Other test classes, mainly because they run multiple tests, instantiate their filter in `__init__()` already.